hera_pspec/pspecdata.py

```diff
@@ -2075,15 +2076,43 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
                     if verbose:
                         print("\n(bl1, bl2) pair: {}\npol: {}".format(blp, tuple(p)))
 
+                    # Check that number of non-zero weight chans >= n_dlys
+                    key1_dof = np.sum(~np.isclose(self.Y(key1).diagonal(), 0.0))
+                    key2_dof = np.sum(~np.isclose(self.Y(key2).diagonal(), 0.0))
+                    if key1_dof < self.spw_Ndlys or key2_dof < self.spw_Ndlys:
+                        if verbose:
+                            print("WARNING: Number of unflagged chans for key1 and/or key2 < n_dlys\n" \
+                                  "which may lead to normalization instabilities.")
+
                     # Build Fisher matrix
-                    if input_data_weight == 'identity' and built_GH:
-                        # in this case, all Gv are the same, so skip if already built for this spw!
-                        pass
+                    if input_data_weight == 'identity':
+                        # in this case, all Gv and Hv differ only by flagging pattern
+                        # so check if we've already computed this
+                        # First: get flag weighting matrices given key1 & key2
+                        Y = np.vstack([self.Y(key1).diagonal(), self.Y(key2).diagonal()])
+
+                        # Second: check cache for Y
+                        matches = [np.isclose(Y, y).all() for y in self._identity_Y.values()]
```

Comment:
This is probably fine, but an alternative would be to hash the array using md5sum.